aesara/tensor/shape.py

```diff
@@ -669,44 +669,32 @@ def infer_shape(self, fgraph, node, ishapes):
             ]
 
     def c_code_cache_version(self):
-        return (8,)
+        return (9,)
 
     def c_code(self, node, name, inputs, outputs, sub):
-        if isinstance(node.inputs[0], TensorVariable):
-            x, shp = inputs
-            (z,) = outputs
-            sdtype = node.inputs[1].type.dtype_specs()[1]
-            fail = sub["fail"]
-            params = sub["params"]
-            return (
-                """
-            assert (PyArray_NDIM(%(shp)s) == 1);
-            npy_intp new_dims[%(params)s->ndim];
-            PyArray_Dims newshape;
-            newshape.ptr = new_dims;
-            newshape.len = %(params)s->ndim;
```

Comment:
Seems like the new code is not using the `ndim` anymore, which the failing test depended on: https://github.com/aesara-devs/aesara/runs/4530278695?check_suite_focus=true#step:6:336

https://github.com/Theano/Theano/pull/5616/files#r102949026